ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/BaseTransaction.java

```diff
@@ -174,9 +174,13 @@ public int getStatus() throws javax.transaction.SystemException
 	public void setTransactionTimeout(int seconds)
 			throws javax.transaction.SystemException
 	{
-		if (seconds >= 0)
+		if (seconds > 0)
+		{
+		    _timeouts.set(Integer.valueOf(seconds));
+		}
+		else if (seconds == 0)
 		{
-		    _timeouts.set(new Integer(seconds));
```

Comment:
FWIW, doing `new Integer(int)` is deprecated in most recent JVMs and ill be removed. To be future-proof, I'd suggest consistently changing all code to use the better `Integer.valueOf(int)` variant.

Reply:
We do that quite a lot so thanks for the tip